### PR TITLE
Add field constraints for emergency contact

### LINKED ISSUE #249: [PE-D][Tester C] Wrong command structure for Import command in UG
format of import command is import fp/<file_path>, instead of the format in the displayed picture

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: hogretup/ped#8

### LINKED ISSUE #253: [PE-D][Tester A] Edit command in UG missing some params (inconsistent)
Params In UG: 
1. n/NAME
2. p/PHONE_NUMBER
3. e/EMAIL
4. a/ADDRESS
5. t/TAG

Params in DMP:
1. n/NAME
2. p/PHONE_NUMBER
3. e/EMAIL
4. a/ADDRESS
5. t/TAG
6. **r/ROOM_NUMBER**
7. **en/EMERGENCY_NAME**
8. **ep/EMERGENCY_PHONE**
9. **g/GRADUATION_YEAR**

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: Domokunx/ped#1

### LINKED ISSUE #259: [PE-D][Tester B] Unable to set or change certain fields easily without having to modify and import an entire file
Emergency contact name, emergency contact phone and graduation year cannot be added through the add command and edit command does not allow users to add on to these fields as well. As a user, I am confused as to how to change or set these fields. 

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: ckaayy/ped#19

### LINKED ISSUE #273: [PE-D][Tester D] Constraints in the wrong place in UG (name)
-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: reidenong/ped#6

### LINKED ISSUE #290: [PE-D][Tester E] UG: incomplete command summary
The export and import commands introduced earlier in the _features_ section are not included in the _command summary_ section.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: wkxcass/ped#13

### LINKED ISSUE #296: Add field constraints for emergency name and phone to UG

### PR DESCRIPTION
**Pull request by charlesliyifeng**
Fixes #296, fixes #273, fixes #259, fixes #253, fixes #290, fixes #249

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -331,11 +333,11 @@ Furthermore, certain edits can cause the DorManagerPro to behave in unexpected w
 
 Action     | Format, Examples
 -----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------
-**Add**    | `add [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/TAG]…​` <br> e.g., `add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+**Add**    | `add [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…​` <br> e.g., `add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
 **Clear**  | `clear`
 **Delete** | `delete INDEX`<br> e.g., `delete 3`
-**Edit**   | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/TAG]…​` <br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`
-**Find**   | `find [n/NAME] [p/PHONE_NUMBER] [r/ROOM_NUMBER] [t/TAG]…​` <br> e.g., `find p/+123 12345 n/Alice Lee r/08-1234 t/friend`
+**Edit**   | `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…​` <br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`
```
Comment (EdaYeo) [inline docs/UserGuide.md:339]: Should we add emergency contact / all the other extra fields here as well?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -407,6 +409,31 @@ Duplicate handling:
 
 ### Emergency contact name
 
+Format: `emergencyName` where `emergencyName` is a sequence of characters corresponding to a student's name
```
Comment (EdaYeo) [inline docs/UserGuide.md:412]: Would it be clearer to just write "A sequence of alphanumeric corresponding to a student's name"? I'm not sure about this so up to you as well.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -358,7 +360,7 @@ Duplicate handling:
 * Two resident student contacts with the same name are not allowed
 > <span style="color:Gray"> NOTE! </span> <br>
 >
-> * Constraint rationale: Phone number constraints are based on the upper and lower limit of country codes, area codes, and number digit lengths.
+> * Constraint rationale: Extremely long names would be difficult to process and display in the profile screen.
```
Comment (EdaYeo) [inline docs/UserGuide.md:363]: Should we also add some constraint rationales for why we only allow "-" for special characters? Though we can let Sean do that too in a different PR since he did the names and I'm not sure what we want to write.

**Review by EdaYeo — COMMENTED** [review]
LGTM, just some questions about certain parts!

File: docs/UserGuide.md [hunk 4]
```diff
@@ -123,15 +123,17 @@ A help window will pop up containing basic introduction to the three core featur
 
 Adds a person to the address book.
 
-Format: `add n/NAME p/PHONE_NUMBER e/EMAIL [r/ROOM_NUMBER] [a/ADDRESS] [t/TAG]...`
+Format: `add n/NAME p/PHONE e/EMAIL [r/ROOM_NUMBER] [a/ADDRESS] [t/TAG]...`
```
[the remark this answers is not in the document]
Comment (charlesliyifeng) [reply docs/UserGuide.md:126]: image

image

I think it mostly displays PHONE in the app, except for the help page?

[the remark this answers is not in the document]
Comment (EdaYeo) [reply docs/UserGuide.md:126]: Oh ok, then we should just change the help page

[the remark this answers is not in the document]
Comment (charlesliyifeng) [reply docs/UserGuide.md:126]: I thought this was decided when I asked for permission to standardise to PHONE in the feature specs?

[the remark this answers is not in the document]
Comment (EdaYeo) [reply docs/UserGuide.md:126]: Yeah, but I noticed it was PHONE_NUMBER in the help and it was too late to change that before the release so I asked again in the group chat and changed to standardise. I think it's better to change back to PHONE though since the rest of the warnings are PHONE as well (which I didn't notice that day)

File: docs/UserGuide.md [hunk 5]
```diff
@@ -331,23 +333,26 @@ Furthermore, certain edits can cause the DorManagerPro to behave in unexpected w
...
-**Find**   | `find [n/NAME] [p/PHONE_NUMBER] [r/ROOM_NUMBER] [t/TAG]…​` <br> e.g., `find p/+123 12345 n/Alice Lee r/08-1234 t/friend`
+**Edit**   | `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [r/ROOM_NUMBER] [en/EMERGENCY_NAME] [ep/EMERGENCY_PHONE] [g/GRADUATION_YEAR] [t/TAG]…​` <br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`
+**Find**   | `find [n/NAME] [p/PHONE] [r/ROOM_NUMBER] [t/TAG]…​` <br> e.g., `find p/+123 12345 n/Alice Lee r/08-1234 t/friend`
 **List**   | `list`
 **Help**   | `help`
 **Clean**  | `clean`
 **Undo**   | `undo`
+**Exit**   | `exit`
+**Export** | `export`
+**Import** | `import f/FILE_PATH` <br> e.g., `import f/./data/SaveFile4.json`
```
Comment (EdaYeo) [inline docs/UserGuide.md:347]: If I'm not wrong, it should be fp/FILE_PATH. I think I noticed a PE-D bug report saying that our UG was wrong for the import prefix. Got to check again though.

**Review by EdaYeo — COMMENTED** [review]
LGTM, just a few more tiny nits and should be good!

**Review by EdaYeo — APPROVED** [review]
LGTM!
